src/sentry/consumer.py

```diff
@@ -1,69 +1,178 @@
+    event_id = data['event_id']
+    cache_key = message['cache_key']
+    start_time = message['start_time']
+    process_task = (
+        store_tasks.process_event_from_reprocessing
+        if message['from_reprocessing']
+        else store_tasks.process_event
+    )
+
+    store_tasks._do_preprocess_event(cache_key, data, start_time, event_id, process_task)
+
+
+def handle_process(message):
+    import sentry.tasks.store as store_tasks
+
+    data = message['data']
+    event_id = data['event_id']
+    cache_key = message['cache_key']
+    start_time = message['start_time']
+
+    if message['from_reprocessing']:
+        task = store_tasks.process_event_from_reprocessing
+    else:
+        task = store_tasks.process_event
+
+    task.delay(cache_key=cache_key, start_time=start_time, event_id=event_id)
+
+
+def handle_save(message):
+    import sentry.tasks.store as store_tasks
+
+    data = message['data']
+    event_id = data['event_id']
+    cache_key = message['cache_key']
+    start_time = message['start_time']
+    project_id = data['project']
+
+    store_tasks._do_save_event(cache_key, data, start_time, event_id, project_id)
+
+
+dispatch = {}
+topic_to_dead_topic_key = {}
+
+
+def handle_task(task):
+    from sentry.utils import json
+
+    if not dispatch:
+        from django.conf import settings
 
-class ConsumerWorker(AbstractBatchWorker):
-    def __init__(self):
-        self.dispatch = {}
         for key, handler in (
-            (settings.KAFKA_PREPROCESS, self.handle_preprocess),
-            (settings.KAFKA_PROCESS, self.handle_process),
-            (settings.KAFKA_SAVE, self.handle_save)
+            (settings.KAFKA_PREPROCESS, handle_preprocess),
+            (settings.KAFKA_PROCESS, handle_process),
+            (settings.KAFKA_SAVE, handle_save)
         ):
             topic = settings.KAFKA_TOPICS[key]['topic']
-            self.dispatch[topic] = handler
-
-    def handle_preprocess(self, message):
-        data = message['data']
-        event_id = data['event_id']
-        cache_key = message['cache_key']
-        start_time = message['start_time']
-        process_task = (
-            store_tasks.process_event_from_reprocessing
-            if message['from_reprocessing']
-            else store_tasks.process_event
+            dispatch[topic] = handler
+            topic_to_dead_topic_key[topic] = key + '-dead'
+
+    topic = task['topic']
+    handler = dispatch[topic]
+
+    try:
+        handler(json.loads(task['value']))
+    except Exception:
+        from sentry.utils import kafka
+
+        topic_config = kafka.get_topic_config(topic)
+        dead_letter_key = topic_config.get('dead-letter-key')
+        if not dead_letter_key:
+            logger.exception(
+                "Error handling message on topic '%s' and no dead-letter-topic is defined.")
+            return
+
+        dead_letter_topic = kafka.get_topic_key_config(dead_letter_key)['topic']
+        logger.exception(
```

Comment:
Not sure if this would create an error feedback loop or if we're using Sentry for Sentry soon (or if settings would disable this becoming a Sentry itself)?